orchestrate.py

```diff
@@ -174,6 +174,35 @@ def post(self):
     def pool(self):
         return self.settings['pool']
 
+class AdminHandler(RequestHandler):
+    def prepare(self):
+        '''
+        Responds with a 401 error if the configured admin auth token is not 
+        present in the request headers.
+        '''
+        if self.admin_token:
+            client_token = self.request.headers.get('Authorization')
+            if client_token != 'token %s' % self.admin_token:
+                app_log.warn('rejecting admin request with token %s', client_token)
+                app_log.info('admin token: %s', self.admin_token)
+                return self.send_error(401)
+        return super(AdminHandler, self).prepare()
+
+    @property
+    def admin_token(self):
+        return self.settings['admin_token']
+
+class APIPoolHandler(AdminHandler):
+    @gen.coroutine
+    def delete(self):
+        '''Drains available containers from the pool.'''
+        yield self.pool.drain()
+        self.set_status(204)
```

Comment:
The RFC for HTTP 204 states "No Content - The server successfully processed the request, but is not returning any content." Personally, I wouldn't tack on this bookkeeping until you need it.